presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -678,6 +691,37 @@ public int getBucketToKeep()
         }
     }
 
+    private static Optional<BucketValidator> createBucketValidator(Path path, Optional<BucketValidation> bucketValidation, OptionalInt bucketNumber, List<ColumnMapping> columnMappings)
+    {
+        return bucketValidation.flatMap(validation -> {
+            Map<Integer, ColumnMapping> baseHiveColumnToBlockIndex = columnMappings.stream()
+                    .filter(mapping -> mapping.getHiveColumnHandle().isBaseColumn())
+                    .collect(toImmutableMap(mapping -> mapping.getHiveColumnHandle().getBaseHiveColumnIndex(), identity()));
+
+            int[] bucketColumnIndices = new int[validation.getBucketColumns().size()];
+
+            List<TypeInfo> bucketColumnTypes = new ArrayList<>();
+            for (int i = 0; i < validation.getBucketColumns().size(); i++) {
+                HiveColumnHandle column = validation.getBucketColumns().get(i);
+                ColumnMapping mapping = baseHiveColumnToBlockIndex.get(column.getBaseHiveColumnIndex());
+                if (mapping == null) {
+                    // the bucket column is not read by the query
```

Comment:
add something from your comment above:

> we don't need to bother with validation since it can't affect the query results

However, is it true actually?
what about filters on `$bucket`?

Reply:
That's an interesting scenario, though technically the bucket filtering is working, as we will only return data in that bucket. If you are using the bucket filter to efficiently read part of the table in different queries, then it will return the correct result. Without reading the bucketing column(s), you can't actually tell that the data is in the wrong bucket. I can't think of such a scenario that would visibly return wrong results.

Reply:
Please maybe capture this as a code comment then.

Reply:
Done